### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by mistermaxis**
### Tasks Completed:
- Add tests for Solver class methods factorial, reverse and fizzbuzz.
- Implement methods factorial, reverse and fizzbuzz for Solver class.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require_relative '../src/solver'
+
+describe Solver do
+  solver = Solver.new
+  context 'When testing the factorial method' do
+    it 'should return the factorial of n' do
+      expect(solver.factorial(0)).to eq 1
+      expect(solver.factorial(5)).to eq 120
+      expect(solver.factorial(6)).to eq 720
+      expect(solver.factorial(7)).to eq 5040
```
Comment (TedLivist) [inline spec/solver_spec.rb:10]: - [ ] Your factorial test is missing a scenario that tests for a negative number, which should raise an exception. Kindly add it.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @mistermaxis,

Good job so far!
✔️ Good documentation
✔️ Rspec is well set up
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
